### DISCUSSION TITLE
Update body's attributes when rendering a fragment

### DISCUSSION THREAD

**Discussion by RomainLanz** [unpoly/unpoly#371] [Q&A] [answered]
Hey there! 👋🏻 

I heavily use data attributes on my `<body>` tag to apply CSS on my page.
When using Unpoly, it does not change the attributes of my `<body>` tag.

I tried to read the incoming fragment to apply the new attributes, but it seems to be already sanitized with the target defined. Therefore I cannot get the new body attributes from here.

How could I do that?

**Comment by adam12** [comment, marked as answer]
So you're targetting something that is _not_ `<body>`, but you want to update `<body>` at the same time? But only the attributes and not the `innerHTML`?

Which event were you listening on?

**Reply by RomainLanz** [reply]
Yes, I am targeting my page's `main` section (since I do not need to get back the sidebar or the header).

Here is an example to understand my issue better.

Let's say we are on the dashboard page that looks like the following:

```html
<html>
<head>
  <!-- ... -->
</head>
<body class="xxx" data-page="dashboard" ...>
  <header>
	<a href="/dashboard up-target>Dashboard</a>
    <a href="/account" up-target>Account</a>
    <!-- ... -->
  </header>

  <aside>
    <!-- ... -->
  </aside>

  <main>
	<h1>Dashboard</h1>
  </main>
</body>
</html>
```

Then, we click on the `Account` link and the server respond with the following HTML:

```html
<html>
<head>
  <!-- ... -->
</head>
<body class="xxx" data-page="account" ...>
  <main>
	<h1>Account</h1>
  </main>
</body>
</html>
```

Unpoly will correctly change the content of `<main>`, but not the `body` attributes (that I need).

I tried using the `up:fragment:inserted` and `up:fragment:loaded` to recover the `body` attributes from the response, but no success so far.

**Reply by triskweline** [reply]
You could move your `[data-page]` attribute to your `<main>` element.

If it absolutely needs to be on the body, you can copy it there with a compiler like this:

```js
up.compiler('main[data-page]', (main) => document.body.dataset.page = main.dataset.page)
```

**Reply by RomainLanz** [reply]
I have moved my attributes to the `<main>` tag, and I will use your compiler if needed, thanks!
